### ISSUE TITLE
Graph Editor: toplevel input `emission_color` is connected but not rendering correctly?

### ISSUE THREAD

**Issue by marwie** [AcademySoftwareFoundation/MaterialX#1445]
Hello,

I've started to learn about MaterialX and am currently reading through the documentation and spec.     

I'm currrently trying to understand if/why my toplevel `<input>` color3 element connected to a `standard_surface/emission_color` input renders as solid white in the MaterialX Graph Editor.

```mtlx
<?xml version="1.0"?>
<materialx version="1.38">
  <input name="myemission" type="color3" value="1, 0, 0" />
  <standard_surface name="standard_surface_surfaceshader" type="surfaceshader">
    <input name="emission_color" type="color3" interfacename="myemission" />
    <input name="emission" type="float" value="1.0" />
  </standard_surface>
  <surfacematerial name="surfacematerial" type="material">
    <input name="surfaceshader" type="surfaceshader" nodename="standard_surface_surfaceshader" />
  </surfacematerial>
</materialx>
```

![image](https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/a3667db3-5c47-48ac-b519-0104ecbf9af3)

Using a constant works as expected

```mtlx
<?xml version="1.0"?>
<materialx version="1.38">
  <constant name="my_emission" type="color3">
    <input name="value" type="color3" value="1, 0, 0" />
  </constant>
  <standard_surface name="standard_surface_surfaceshader" type="surfaceshader">
    <input name="emission_color" type="color3" nodename="my_emission" />
    <input name="emission" type="float" value="1.0" />
  </standard_surface>
  <surfacematerial name="surfacematerial" type="material">
    <input name="surfaceshader" type="surfaceshader" nodename="standard_surface_surfaceshader" />
  </surfacematerial>
</materialx>
```

![image](https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/61b8b5df-b81c-4b9a-84c3-0833c74144ee)

**Comment by kwokcb** [comment]
Hi @marwie ,

This is a known issue for code generation (top level input driver), but good to know this is a simple reoccurring workflow so can check on prioritization. Inputs evaluate properly when inside a node graph as a workaround.

There is no “group selected nodes and make into a graph” yet, so need to start with a graph. 

Look forward to any additional questions / feedback on this.

Thanks.

**Comment by kwokcb** [comment]
Good to know. I know it’s a bit of a steep curve getting up to speed, and there are spec vs implementation caveats that probably aren’t that clearly stated.

**Comment by marwie** [comment]
I might have a follow up issue / question. If I understood you correctly the input inside the nodegraph should work and I should be able to pipe it through an output to connect it with some `standard_surface/input`. But it doesn't seem to work without adding *some* node inbetween (like a constant node). Perhaps because nothing is still triggering the evaluate for the input (even when nested) or I misunderstood. 

I recorded a short video below.   
There seem to be some other weird things going on as well in there (like having an output element connected to two other elements)

https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/c401ed55-4a05-4aea-a682-cfbaab8c10ca

**Comment by kwokcb** [comment]
Hi @marwie,
For the direct connection, I think this is considered to be invalid but it's not stated explicitly in the spec. We end up with validation errors currently. The editor seems to be allowing this invalid case so you the weird fan-in case.

Can you pull this out into a separate issue? ( and CC dbsmythe and jstone-lucasfilm for spec input).

**Comment by jstone-lucasfilm** [comment]
Thanks for this report, @marwie, and thanks to @kwokcb for the fix in pull request #1453.
